modules/users/client/controllers/settings/manage-social-accounts.client.controller.js

```diff
@@ -42,5 +43,15 @@
     function onRemoveSocialAccountError(response) {
       Notification.error({ message: response.message, title: '<i class="glyphicon glyphicon-remove"></i> Remove failed!' });
     }
+
+    // OAuth provider request
+    function callOauthProvider(url) {
+      if ($state.previous && $state.previous.href) {
+        url += '?redirect_to=' + encodeURIComponent($state.$current.url.prefix);
```

Comment:
Using `prefix` does work, so I'm not questioning that. You're right though, that I was getting the two confused, and even so it wouldn't work to use `$state.current.url` for the exact reason you stated; it would return `/accounts`.

My confusion was reinforced by the fact that using `$state.$current.url` would work in this case due to the URI encoding converting the `url` property to a string, which returns `/settings/accounts`. 

All that aside, you are correct & using `prefix` is ok. It just wasn't clear as to why it was used. The only thing I would request is to remove the conditional statement since it's incorrect & it's not needed.